docs/usage/self-hosted-configuration.md

```diff
@@ -59,6 +59,46 @@ But before you disable templating completely, try the `allowedPostUpgradeCommand
 
 ## allowScripts
 
+## allowedHeaders
+
+List of headers that are allowed to be forwarded to the HTTP request. This is useful when the `registryUrlTemplate` employs a specific authentication system. By default, all headers starting with "X-" are allowed, but you can include additional headers using this option. If declared, it will override the default "X-" allowed headers. `allowedHeaders` is an array of minimatch-compatible globs or re2-compatible regex represented as strings.
```

Comment:
```suggestion
This is option particularly useful when a registry employs a specific authentication system not already covered by Renovate's standard credential handling in `hostRules`.
By default, all headers starting with "X-" are allowed, but you can permit additional headers using this option.
If declared, it will override the default "X-" allowed headers, so you should include them in your config if you wish for them to remain allowed.
`allowedHeaders` is an array of minimatch-compatible globs or re2-compatible regex strings.
```